numpy/core/tests/test_multiarray.py

```diff
@@ -5057,46 +5060,61 @@ def dup_bigint(fd):
 
         old_dup = os.dup
         try:
-            with open(self.filename, 'wb') as f:
-                self.x.tofile(f)
+            with open(tmp_filename, 'wb') as f:
+                x.tofile(f)
                 for dup, exc in ((dup_str, TypeError), (dup_bigint, OSError)):
                     os.dup = dup
                     assert_raises(exc, np.fromfile, f)
         finally:
             os.dup = old_dup
 
-    def _check_from(self, s, value, **kw):
+    def _check_from(self, s, value, filename, **kw):
         if 'sep' not in kw:
             y = np.frombuffer(s, **kw)
         else:
             y = np.fromstring(s, **kw)
         assert_array_equal(y, value)
 
-        with open(self.filename, 'wb') as f:
+        with open(filename, 'wb') as f:
             f.write(s)
-        y = np.fromfile(self.filename, **kw)
+        y = np.fromfile(filename, **kw)
         assert_array_equal(y, value)
 
-    def test_nan(self):
+    @pytest.fixture(params=[True, False])
+    def comma_decimal_point_locale(self, request):
+        if request.param:
+            with CommaDecimalPointLocale():
+                yield
+        else:
+            yield
+
+    def test_nan(self, tmp_filename, comma_decimal_point_locale):
```

Comment:
It is not clear form the code how comma_decimal_point_locale is ever used. A comment would be helpful.

Reply:
just to be clear: the previous code was also opaque as to what this does